### ISSUE TITLE
Support `reshape` on the list and arr namespaces

### ISSUE THREAD

**Issue by thomasaarholt** [pola-rs/polars#15724]
**TLDR: Please support** 

```python
pl.col("col").list.reshape
#                  ^^^^^^^ missing this
pl.col("col").arr.reshape
#                 ^^^^^^^ missing this
```
### Description
[In Discord today I helped someone with a query](https://discord.com/channels/908022250106667068/1229816436478967941/1230143639661838407). 

They desired to reshape a list of guaranteed even length to a nested list of 2-length lists.

```python
┌────────────────────┐
│ list_column        │
│ ---                │
│ list[i64]          │
╞════════════════════╡
│ [1, 2, 3, 4]       │
│ [1, 2, 3, 4, 5, 6] │
└────────────────────┘
```
to
```python
┌──────────────────────────┐
│ list_column              │
│ ---                      │
│ list[list[i64]]          │
╞══════════════════════════╡
│ [[1, 2], [3, 4]]         │
│ [[1, 2], [3, 4], [5, 6]] │
└──────────────────────────┘
```

I first looked for a reshape method on the list namespace (`pl.col("list_column").list.reshape`), since my desire was to reshape the list from one shape to another. Then I realized that `reshape` only exists on the `col` namespace and applies to the entire column. My workaround was to treat a row as a column by using the `list.eval` method:

```python
df = pl.DataFrame(
    {
        "list_column": [
            [1, 2, 3, 4],
            [1, 2, 3, 4, 5, 6],
        ]
    }
)

df.with_columns(pl.col("list_column").list.eval(pl.element().reshape((-1, 2))))
```
I would have expected:
```python
df.with_columns(pl.col("list_column").list.reshape((-1, 2)))
```
It would be nice to provide this method on both the list and array namespaces. While `array` should "obviously" support this method, since `reshape` supports a `-1` value to infer the dimension size, it makes sense to also provide this on the `list` namespace, as shown to be useful in my example above.

**Comment by choucavalier** [comment]
agreed. this would be great especially in the context of machine learning where we need to create arrays of the right shape

**Comment by coastalwhite** [comment]
I think I am going to close this in favor of #21302.
